### PR TITLE
crimson/os: Don't limit the amount of returned keys per omap get call

### PR DESCRIPTION
**Pull request by AmnonHanuhov**
Users of `FuturizedStore::omap_get_values()` expect to get all the keys
that are currently in the omap, but behind the scenes we were limiting
`CyanStore::omap_get_values()` to return only a hardcoded number of values,
specifically `MAX_KEYS_PER_OMAP_GET_CALL` which is 32.

Signed-off-by: Amnon Hanuhov <AmnonSWE@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/alienstore/alien_store.cc [hunk 1]
```diff
@@ -396,7 +396,8 @@ auto AlienStore::omap_get_values(CollectionRef ch,
 
 auto AlienStore::omap_get_values(CollectionRef ch,
                                  const ghobject_t &oid,
-                                 const std::optional<string> &start)
+                                 const std::optional<string> &start,
+                                 const uint64_t max_returned_vals)
```
Comment (rzarzynski) [inline src/crimson/os/alienstore/alien_store.cc:400]: How do you want to implement that in `AlienStore`?

**Comment by rzarzynski** [comment]
jenkins retest this please

**Review by rzarzynski — APPROVED** [review]
Yeah, the max-entries-per-request logic is ignored by other implementations of `FuturizedStore` and in `CyanStore` it's broken. It makes sense to drop it from there and properly implement on an upper layer (created [a Trello card](https://trello.com/c/6mC9Grw5/117-implement-osdmaxomapentriesperrequest) to track this).

**Comment by rzarzynski** [comment]
@athanatos: want to take a look before merge?

**Review by athanatos — APPROVED** [review]

**Pull request merged**